RadarSDK/RadarLogBuffer.m

```diff
@@ -7,24 +7,46 @@
 
 #import "RadarLogBuffer.h"
 #import "RadarLog.h"
+#import "RadarFileStorage.h"
+#import "RadarSettings.h"
 
+static const int MAX_PERSISTED_BUFFER_SIZE = 500;
+static const int MAX_MEMORY_BUFFER_SIZE = 200;
+static const int PURGE_AMOUNT = 250;
 static const int MAX_BUFFER_SIZE = 500;
-static const int PURGE_AMOUNT = 200;
 
 static NSString *const kPurgedLogLine = @"----- purged oldest logs -----";
 
+static int counter = 0;
+
 @implementation RadarLogBuffer {
     NSMutableArray<RadarLog *> *mutableLogBuffer;
+    NSMutableArray<RadarLog *> *inMemoryLogBuffer;
 }
 
 - (instancetype)init {
     self = [super init];
     if (self) {
+        _featureFlag = [RadarSettings featureSettings].useLogPersistence;
         mutableLogBuffer = [NSMutableArray<RadarLog *> new];
+        inMemoryLogBuffer = [NSMutableArray<RadarLog *> new];
+        
+        NSString *documentsDirectory = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES).firstObject;
+        self.logFileDir = [documentsDirectory stringByAppendingPathComponent:@"radar_logs"];
+        if (![[NSFileManager defaultManager] fileExistsAtPath:self.logFileDir isDirectory:nil]) {
+            [[NSFileManager defaultManager] createDirectoryAtPath:self.logFileDir withIntermediateDirectories:YES attributes:nil error:nil];
+        }
+        self.fileHandler = [[RadarFileStorage alloc] init];
+        _timer = [NSTimer scheduledTimerWithTimeInterval:20.0 target:self selector:@selector(persistLogs) userInfo:nil repeats:YES];
+        
     }
     return self;
 }
 
+- (void)setFeatureFlag:(BOOL)featureFlag {
+    _featureFlag = featureFlag;
+}
+
```

Comment:
this is the right approach

the name suggests something more general — I think a more descriptive name could be helpful

Reply:
`enablePersistentLog` ?

Reply:
yes, that's better